### PR TITLE
[SITES-472] Authorization fixes

### PR DESCRIPTION
**Pull request by tessereth**
This includes refactoring the auth tests so it's a little easier to extend for new routes.

### CODE REVIEW

File: app/controllers/editorial/news_controller.rb [hunk 1]
```diff
@@ -12,10 +12,11 @@ def show
         section: params[:section],
         slug: params[:slug]
       )
-
+      authorize! :read, @article
     end
 
     def new
+      authorize! :create, :news
```
Comment (ltankey) [inline app/controllers/editorial/news_controller.rb]: Would it be more DRY to use a `before` hook?

File: app/controllers/editorial/submissions_controller.rb [hunk 2]
```diff
@@ -19,6 +19,7 @@ def index
     end
 
     def create
+      authorize! :update, @node
```
Comment (ltankey) [inline app/controllers/editorial/submissions_controller.rb]: Again, RE: DRYing this up, perhaps we could use a before call.

**Comment by tessereth** [comment]
update changelog post 0.7.1 release

**Comment by ltankey** [comment]
These tests are much better - we're missing the `:owner` role and requests list but otherwise this is a great improvement.

LGTM

**Comment by tessereth** [comment]
@ltankey There is no request list page. Or rather the route is there but it displays the editorial index page (probably a byproduct of our controller hierarchy). We should really remove that route until it's implemented.

**Comment by micapam** [comment]
LGTM. Nice use of rspec shared examples

**Comment by tessereth** [comment]
Updated to include topic owner and the request show route in the tests. Also update routes.rb not to route to request pages that don't exist.

**Pull request merged**